From 8b59d7c91a2f9d03296aaf00295af2ea803d337c Mon Sep 17 00:00:00 2001
From: Torge Matthies <openglfreak@googlemail.com>
Date: Tue, 19 Mar 2024 10:19:09 +0100
Subject: [PATCH 14/43] winegstreamer: Fixate caps in autoplug_continue_cb.

Not strictly necessary but avoids an assert in gstreamer code.
---
 dlls/winegstreamer/wg_parser.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/dlls/winegstreamer/wg_parser.c b/dlls/winegstreamer/wg_parser.c
index 60d6d2a669e..1ff4da1971c 100644
--- a/dlls/winegstreamer/wg_parser.c
+++ b/dlls/winegstreamer/wg_parser.c
@@ -527,7 +527,9 @@ static gboolean autoplug_continue_cb(GstElement * decodebin, GstPad *pad, GstCap
 {
     struct wg_format format;
 
+    caps = gst_caps_fixate(gst_caps_copy_nth(caps, 0));
     wg_format_from_caps(&format, caps);
+    gst_caps_unref(caps);
 
     return !format_is_compressed(&format);
 }
-- 
2.45.2

